OSDOCS-6652 AWS shared private hosted zones (#64114)
openshift/openshift-docs: modules/installation-aws-permissions-iam-shared-vpc.adoc

// Module included in the following assembly:
// * installing/installing_aws/installing-aws-account.adoc

:_content-type: REFERENCE
[id="installation-aws-permissions-iam-shared-vpc_{context}"]

> = Trust policies for installing into a shared VPC

Author: I'm fine with moving this into the AWS VPC installation docs, but I don't want to move it into the installation configuration parameters table because I think it will be hard to find, and it will take up a lot of space in the table.

// Module included in the following assembly:
// * installing/installing_aws/installing-aws-account.adoc

:_content-type: REFERENCE
[id="installation-aws-permissions-iam-shared-vpc_{context}"]

> = Trust policy for installing into a shared VPC

Reviewer: @bscott-rh Great update in this section, it's clear.
one comment is about the title, as permission policy was added into this section, so this part contains _trust policy + permission policy_, how about
```suggestion
= The role policy for installing into a shared VPC
```

@patrickdillon WDYT?

Author: Thanks, I have slightly reworded the title with this in mind.

> If you install your cluster using a shared VPC, you can use the `Passthrough` or `Manual` credentials mode. You must add the cluster owner account as a principal in the trust policy of the account that owns the VPC.

Reviewer 2: ```suggestion
If you install your cluster using a shared VPC, you can use the `Passthrough` or `Manual` credentials mode. You must add the Installer IAM role from the cluster owner account as a principal in the trust policy of the account that owns the VPC.
```

Author: Good suggestion, thanks

> If you use `Passthrough` mode, add the Amazon Resource Name (ARN) of the account that creates the cluster (for example `arn:aws:iam::123456789012:user/clustercreator`) to the trust policy as a principal. 

Reviewer 3: ISG says to try and avoid parentheses in general text:
```suggestion
If you use `Passthrough` mode, add the Amazon Resource Name (ARN) of the account that creates the cluster, such as `arn:aws:iam::123456789012:user/clustercreator`, to the trust policy as a principal. 
```

> If you use `Manual` mode, add the ARN of the account that creates the cluster as well as the ARN of the ingress operator role in the cluster owner account (for example `arn:aws:iam::123456789012:role/<cluster-name>-openshift-ingress-operator-cloud-credentials`) to the trust policy as principals.

Reviewer 3: same comment as above

> The following actions must be added to the policy:

Reviewer 3: ISG recommends active voice and second person pov:
```suggestion
You must add the following actions to the policy:
```

> .Required trust policy actions for shared VPC installation

Reviewer: ```suggestion
.Required permission policy actions for shared VPC installation
```
